### PR TITLE
Support for custom uri's for non-fungible token pools

### PR DESCRIPTION
**Pull request by shorsher**
in a chain with #80

### CODE REVIEW

File: samples/solidity/contracts/ERC1155MixedFungible.sol [hunk 1]
```diff
@@ -30,16 +30,23 @@ import "@openzeppelin/contracts/utils/Context.sol";
...
+    uint256 constant TYPE_MASK = type(uint128).max << 128;
+    uint256 constant NF_INDEX_MASK = type(uint128).max;
     uint256 constant TYPE_NF_BIT = 1 << 255;
 
     uint256 nonce;
     mapping (uint256 => address) public creators;
     mapping (uint256 => uint256) public maxIndex;
 
+    // inherited ERC1155 `_uri` is private, so need our own within this contract
+    string private _baseTokenURI;
```
[the remark this answers is not in the document]
Comment (shorsher) [reply samples/solidity/contracts/ERC1155MixedFungible.sol]: yep, that's a great catch. Thank you!

File: samples/solidity/contracts/InterfaceCheck.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+import './IERC1155MixedFungible.sol';
+
+/**
+ * Test utility for checking ERC165 interface identifiers.
+ */
+contract InterfaceCheck {
```
Comment (shorsher) [inline samples/solidity/contracts/InterfaceCheck.sol:10]: After looking more at the connector, it seems to me we don't even need to support ERC165. If I am understanding the code currently, the 1155 connector does not allow creating pools from existing contracts on chain.

File: samples/solidity/contracts/ERC1155MixedFungible.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (shorsher) [reply samples/solidity/contracts/ERC1155MixedFungible.sol]: After further consideration, I think having a base URI across all NFT's is OK for the time being. It is already the default behavior for the 1155 contract. It’s in the format `firefly://token/{id}` and the connector itself is responsible for replacing {id} with something meaningful.

[the remark this answers is not in the document]
Comment (shorsher) [reply samples/solidity/contracts/ERC1155MixedFungible.sol]: There may be additional implications across the firefly ecosystem to change this, I know the CLI already deploys the contract with a base URI.

[the remark this answers is not in the document]
Comment (jimthematrix) [reply samples/solidity/contracts/ERC1155MixedFungible.sol]: that makes sense, forgot that this was the pattern proposed in the EIP. I'm OK with going with that.

**Review by jimthematrix — APPROVED** [review]

**Review by nguyer — APPROVED** [review]

File: samples/solidity/contracts/ERC1155MixedFungible.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (shorsher) [reply samples/solidity/contracts/ERC1155MixedFungible.sol]: thank you!

**Pull request merged**